lib/senders/sms.js

```diff
@@ -69,6 +81,50 @@ module.exports = (log, translator, templates, config) => {
     }
   }
 
+  function pollCurrentSpend () {
+    let limit
+
+    sns.getSMSAttributes({ attributes: [ 'MonthlySpendLimit' ] }).promise()
+      .then(result => {
+        limit = parseFloat(result.MonthlySpendLimit)
+        if (isNaN(limit)) {
+          throw new Error(`Invalid MonthlySpendLimit "${result.MonthlySpendLimit}"`)
+        }
+
+        const now = new Date()
+        return cloudwatch.getMetricsStatistics({
+          Namespace: 'AWS/SNS',
+          MetricName: 'SMSMonthToDateSpentUSD',
+          StartTime: startOfMonth(now).toISOString(),
+          EndTime: now.toISOString(),
+          Period: SECONDS_PER_WEEK,
+          Statistics: [ 'Sum' ]
+        }).promise()
+      })
+      .then(result => {
+        const { Datapoints: data } = result
+
+        if (! Array.isArray(data)) {
+          throw new Error('Invalid Datapoints')
+        }
+
+        const current = data.reduce((sum, datum) => sum + parseFloat(datum.Sum), 0)
+
+        if (isNaN(current)) {
+          throw new Error('Invalid Datapoints')
+        }
+
+        isBudgetOk = current <= limit - CREDIT_THRESHOLD
+      })
+      .catch(err => {
+        log.error({ op: 'sms.budget.error', err: err.message })
+
+        // If we failed to query the data, assume current spend is fine
```

Comment:
I wonder how often this error would occur and if we should be defensive about it and disable SMS. Probably not for this PR, but we could disable it after X amount of polling fails.

Reply:
Hmm, I don't think we'd want to disable SMS in that case. Essentially if this automation fails we want to go back to the current scenario, where we assume SMS is working fine but can disable it in config if we need to. Forcing it disabled here would deny us that.

And errors here might happen for any reason of course, completely unrelated to SES. Maybe we update to a new version of `aws-sdk` and it conflicts with our code somehow. Or maybe there's some kind of outage that affects Cloudwatch but not SES. In cases like that, disabling SMS would actually be *less* defensive.

Reply:
I agree with @philbooth here